src/extension.ts

```diff
@@ -74,6 +75,9 @@ export function activate(context: vs.ExtensionContext) {
 	context.subscriptions.push(vs.languages.registerDocumentHighlightProvider(DART_MODE, new DartDocumentHighlightProvider(analyzer)));
 	context.subscriptions.push(new AnalyzerStatusReporter(analyzer));
 
+	// Set up commands for Dart editors.
+	new DartCommands(context, analyzer);
```

Comment:
It feels really weird calling a constructor and not assigning the result, and I think I've tried to make sure only `extension.ts` is doing things like adding to `context.subscriptions`.

I think it'd be better to make `DartCommands` disposable and put it into `context.subscriptions` here and have its dispose handle its children, or make this a function (`registerCommands`?) that sets them up and returns disposables that are put into the collection here.

(Happy to discuss if you think the idea is weird though!)